tests/test_gil_scoped.py

```diff
@@ -97,3 +97,9 @@ def test_python_to_cpp_to_python_from_process():
 def test_cross_module_gil():
     """Makes sure that the GIL can be acquired by another module from a GIL-released state."""
     m.test_cross_module_gil()  # Should not raise a SIGSEGV
+
+
+# Checks for segfault
+def test_potential_segfault():
```

Comment:
I could port just this test to `unittest`, so that I can run it with our sanitizers (ASAN, MSAN, TSAN).
Does that sound like a useful idea?

Reply:
Yes yes yes! Just running it in plain Python does cause a segfault ~50% of the time (possibly regardless of the number of times it runs). There are no actual asserts here, so should be trivial to run.

Reply:
(The single file I posted in the issue fails by itself if run directly; you just need to put it in the right place to load the cpp output module).